documentation/docs/developers/connectors/task4.md

```diff
@@ -1,8 +1,178 @@
-# Task 4 — Configuration & Additional Functions
+# Task 4 — API Throttler, Configurations & Additional Functions
 
-This section explains the configuration for the required file, functions, and other optional settings to integrate your connector with Hummingbot.
+This section explains the steps required to integrate the API throttler into the connector as well as detailing the necessary configurations to integrate your connector with Hummingbot.
 
-## Configuration
+## API Throttler
+
+This section will detail the necessary steps to integrate the `AsyncThrottler` into the connector. 
+The `AsyncThrottler` class utilizes asynchrounous context managers to throttle API and/or WebSocket requests.
+
+!!! note
+    The integration of the `AsyncThrottler` into the connector is entirely optional, but it is recommended to enable a better user experience as well as allowing users to manually configure the usable rate limits per Hummingbot instance.
+
+### Types of Rate Limits
+
+There are several types of rate limits that can be handled by the `AsyncThrottler` class. The following sections will detail(with examples) how to initialize the necessary `RateLimit` and how to the throttler is consumed by the connector for each of the different rate limit types. 
+
+!!! warning
+    It is important to identify the exchange's rate limit implementation.
+
+#### 1. Rate Limit per endpoint
+
+This refers to rate limits that are applied on a per endpoint basis. For this rate limit type, the key information to retrieve for each endpoint would be its assigned **limit** and **time interval**.
```

Comment:
I think it would also be nice to give an example of how this works. Something like:

The time interval is on a rolling basis. For example, if the endpoint's rate limit is 20 and the time interval is 60, the next time the throttler is called for that endpoint, it will check if there have been 20 calls made in the past minute from the current moment.